transformer/metadata.go

```diff
@@ -75,11 +75,67 @@ func (meta *Metadata) Transform(c *skogul.Container) error {
 	return nil
 }
 
+// flattenStructure copies a nested object/array to the root level
+func flattenStructure(nestedPath []string, metric *skogul.Metric) error {
+	newPath := nestedPath[0]
+
+	if len(nestedPath) > 1 {
+		for _, p := range nestedPath[1:] {
+			newPath = fmt.Sprintf("%s__%s", newPath, p)
+		}
+	}
+
+	obj, err := skogul.ExtractNestedObject(metric.Data, nestedPath)
+
+	if err == nil {
+		nestedObj, ok := obj[nestedPath[len(nestedPath)-1]].(map[string]interface{})
+
+		if !ok {
+			fmt.Println("Failed to cast to map[str]intf, trying []interface{} instead")
+
+			nestedObjArray, ok := obj[nestedPath[len(nestedPath)-1]].([]interface{})
+			if !ok {
+				fmt.Println("Failed to cast to []interface{}, aborting transform")
+				return skogul.Error{Reason: "Failed cast"}
+			}
+
+			nestedObj = make(map[string]interface{})
+			for i, val := range nestedObjArray {
+				fmt.Printf("writing arr  i '%s' to new: %d = %+v\n", fmt.Sprintf("%d", i), i, val)
+
+				obj, isMap := val.(map[string]interface{})
+
+				// If the cast is successful, the array of items is a list of map[string]interface{},
+				// and we want to extract each key to its own key in the root, prefixed with the path
+				// Otherwise, the array is a list of a primitive construct and we
+				// simply prefix the key with the array index
+				if isMap {
+					for key, val := range obj {
+						nestedObj[fmt.Sprintf("%d__%s", i, key)] = val
```

Comment:
The separator should probably be configurable. I think a lot of people would like/prefer a dot, for example. Let's make that version 2.